### ISSUE TITLE
Warning: Ignoring InnerClasses attribute for an anonymous inner class ...

### ISSUE THREAD

**Issue by GoogleCodeExporter** [josaluntu/ksoap2-android#71]
```
When compiling an Android project with 
ksoap2-android-assembly-2.5.7-jar-with-dependencies.jar as a referenced library 
(external jar) I get this warning:

Ignoring InnerClasses attribute for an anonymous inner class
(org.ksoap2.transport.KeepAliveHttpsTransportSE$1) that doesn't come with an
associated EnclosingMethod attribute. This class was probably produced by a
compiler that did not target the modern .class file format. The recommended
solution is to recompile the class from source, using an up-to-date compiler
and without specifying any "-target" type options. The consequence of ignoring
this warning is that reflective operations on this class will incorrectly
indicate that it is *not* an inner class.


What version of the product are you using? On what operating system?
2.5.7, earlier 2.5.4 and 2.5.2 with the same problem.
```

Original issue reported on code.google.com by `elias.ny...@gmail.com` on 25 Jul 2011 at 11:36

**Comment by GoogleCodeExporter** [comment]
```
ksoap2 is compiled with backward compatibility for java 1.3 for j2me support 
and therefore the error message it probably correct. However it is just a 
warning and should not have any problems associated for normal use cases. I 
will see if can can fix it somehow anyway.
```

Original comment by `mosa...@gmail.com` on 25 Jul 2011 at 8:17
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
Original comment by `mosa...@gmail.com` on 5 Aug 2011 at 6:04

**Comment by GoogleCodeExporter** [comment]
```
When compiling ksoap2-android with source and target jdk version to 1.5 rather 
than 1.3 this problem disappears so the error message is correct. Looking at 
code I yet have to find a fix for 1.3.. 
```

Original comment by `mosa...@gmail.com` on 5 Aug 2011 at 8:07

**Comment by GoogleCodeExporter** [comment]
```
There are no problem occurring at runtime due to that warning. Everything works 
fine. 
```

Original comment by `mosa...@gmail.com` on 14 Sep 2011 at 9:02

**Comment by GoogleCodeExporter** [comment]
```
Maybe you could split the codebase into 2 branches - one that is complied for 
the older versions (which may or may not be maintained), and a Current branch, 
which is compiled for the modern versions, which is maintained.  Seems like 
that would fix the problem.
```

Original comment by `cgtyo...@gmail.com` on 25 Sep 2011 at 1:22

**Comment by GoogleCodeExporter** [comment]
```
I could do that but there is really no issue. The warning is a red herring and 
there is NO problem resulting from it whatsoever. Just dont worry about it ... 
I might upgrade the project to Java 5 later ... 
```

Original comment by `mosa...@gmail.com` on 26 Sep 2011 at 4:00

**Comment by GoogleCodeExporter** [comment]
```
It's a really big problem for me because I must compile with absolutely no 
warnings ! (for certification reasons)
```

Original comment by `hadrien....@gmail.com` on 18 Nov 2011 at 6:28

**Comment by GoogleCodeExporter** [comment]
```
Just compile your own build with java 5 then.. you just have to change th 
compiler plugin configuration in the top level pom.
```

Original comment by `mosa...@gmail.com` on 18 Nov 2011 at 6:37

**Comment by GoogleCodeExporter** [comment]
```
A fix has been contributed and the upcoming 2.6.4 release will contain it. Also 
see https://github.com/mosabua/ksoap2-android/pull/16
```

Original comment by `mosa...@gmail.com` on 30 Apr 2012 at 6:13
- Changed state: **Fixed**
